docs/04-resources/04-tutorials/rootstock-go.md

```diff
@@ -0,0 +1,490 @@
+
+
+```solidity
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+contract Storage {
+    uint256 number;
+
+    function store(uint256 num) public {
+        number = num;
+    }
+
+    function retrieve() public view returns (uint256){
+        return number;
+    }
+}
+```
+There are two functions in the contract, one to store a number on the blockchain, `function store` (changes state, involves a transaction), another to retrieve the stored number `function retrieve`(no transaction).
+The ABI is provided below.
+```javascript
+[
+  {
+    "type": "function",
+    "name": "retrieve",
+    "inputs": [],
+    "outputs": [
+      {
+        "name": "",
+        "type": "uint256",
+        "internalType": "uint256"
+      }
+    ],
+    "stateMutability": "view"
+  },
+  {
+    "type": "function",
+    "name": "store",
+    "inputs": [
+      {
+        "name": "num",
+        "type": "uint256",
+        "internalType": "uint256"
+      }
+    ],
+    "outputs": [],
+    "stateMutability": "nonpayable"
+  }
+]
+```
+
+### Read from the Contract
+Paste the code below into `main.go`.
+```go
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"math/big"
+	"strings"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+const storageABI = `[
+  {
+    "type": "function",
+    "name": "retrieve",
+    "inputs": [],
+    "outputs": [
+      {
+        "name": "",
+        "type": "uint256",
+        "internalType": "uint256"
+      }
+    ],
+    "stateMutability": "view"
+  },
+  {
+    "type": "function",
+    "name": "store",
+    "inputs": [
+      {
+        "name": "num",
+        "type": "uint256",
+        "internalType": "uint256"
+      }
+    ],
+    "outputs": [],
+    "stateMutability": "nonpayable"
+  }
+]`
+
+func main() {
+	rpcURL := "https://rpc.testnet.rootstock.io/YOUR-API-KEY"
+
+	client, err := ethclient.Dial(rpcURL)
+	if err != nil {
+		log.Fatal("Failed to connect to Ethereum node:", err)
+	}
+	defer client.Close()
+
+	contractAddress := common.HexToAddress("0x8b1Fc84e39B396528431FF97010bF51e458E202d")
+
+	parsedABI, err := abi.JSON(strings.NewReader(storageABI))
+	if err != nil {
+		log.Fatal("Failed to parse ABI:", err)
+	}
+
+	callData, err := parsedABI.Pack("retrieve")
+	if err != nil {
+		log.Fatal("Failed to pack retrieve call:", err)
+	}
+
+	msg := ethereum.CallMsg{
+		To:   &contractAddress,
+		Data: callData,
+	}
+	result, err := client.CallContract(context.Background(), msg, nil)
+	if err != nil {
+		log.Fatal("Failed to call contract:", err)
+	}
+
+	output, err := parsedABI.Unpack("retrieve", result)
+	if err != nil {
+		log.Fatal("Failed to unpack result:", err)
+	}
+
+	storedNumber := output[0].(*big.Int)
+	fmt.Println("Stored Number:", storedNumber)
+}
+```
+```bash
+go run main.go
+
+Stored Number: 3
```

Comment:
Kindly remove the response inside the code block as this may be confusing to the reader copying the code block. 

Response:
Stored Number: 3